lib/OpenQA/Utils.pm

```diff
@@ -111,20 +119,28 @@ sub needle_info($$$$) {
     }
 
     my $needle;
-    eval { $needle = decode_json(<$JF>); };
-    close($JF);
-
-    if ($@) {
-        warn "failed to parse $needledir/$name.json: $@";
-        return;
+    # try-catch actually resolves functions, so we can not "return from needle_info"
+    #   while inside the catch clause
+    # thus `return unless` aborts the needle_info method when the `return undef` in
+    #   the catch clause is evaluated
+    return unless try {
+        $needle = decode_json(<$JF>);
+        return 1;
+    }
+    catch {
+        warn "failed to parse $fn: $_";
+        return 0;
     }
+    finally {
+        close($JF);
+    };
```

Comment:
One time I actually read the Perl Docs *sigh*
But sure, this is obviously much more reasonable approach. Update is on the way.

Thanks for the review!